src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquirerTestHooks.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Identity.Client;
+
+namespace Microsoft.Identity.Web.TestOnly
+{
+    /// <summary>
+    /// INTERNAL – intended solely for unit-tests.
+    /// Allows tests to inject their own IMsalHttpClientFactory so that
+    /// Managed-Identity flows use mocked HTTP Client Factory.
+    /// </summary>
+    internal static class TokenAcquirerTestHooks
+    {
+        /// <remarks>
+        /// Tests call this *before* the first MI token request.
+        /// Product code never sets this value.
+        /// </remarks>
+        internal static void UseTestHttpClientFactory(IMsalHttpClientFactory factory)
+            => HttpClientFactoryOverride = factory;
+
+        internal static IMsalHttpClientFactory? HttpClientFactoryOverride { get; private set; }
```

Comment:
@gladjohn - I had a chat with @jmprieur and the recommendation is to use depedency injection instead. You can have an `internal` interface and so the test injects an internal object (the httpclient factory).

Reply:
![image](https://github.com/user-attachments/assets/24e9c74f-e447-4e6e-aa6b-8a599a7a5abf)

Reply:
Thanks @bgavrilMS @jmprieur, let me take at it.

Reply:
modified